pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -320,20 +312,12 @@ def __getstate__(self, drop_value: bool = False) -> dict:
         def getstate(item: ResultMetric) -> dict:
             return item.__getstate__(drop_value=drop_value)
 
-        items = apply_to_collection(dict(self), (ResultMetric, ResultMetricCollection), getstate)
+        items = apply_to_collection(dict(self), ResultMetric, getstate)
         return {"items": items, "meta": self.meta.__getstate__(), "_class": self.__class__.__name__}
 
     def __setstate__(self, state: dict, sync_fn: Optional[Callable] = None) -> None:
-        def setstate(item: dict) -> Union[Dict[str, ResultMetric], ResultMetric, Any]:
-            # recurse through dictionaries to set the state. can't use `apply_to_collection`
-            # as it does not recurse items of the same type.
-            if not isinstance(item, dict):
-                return item
-            if item.get("_class") == ResultMetric.__name__:
-                return ResultMetric._reconstruct(item, sync_fn=sync_fn)
-            return {k: setstate(v) for k, v in item.items()}
-
-        items = setstate(state["items"])
```

Comment:
This code changed because we don't support recursively nested `ResultMetricCollections` (we were going to)

And mypy has bad support for recursion

Reply:
Yes, I think we won't support it. Better to have minimalist support there and cleaner code.